# **Week 8 Project**

# **Explanation of the Project:**
# Dataset: Life Expectancy at Birth Across the Globe

### This project looks into global trends in life expectancy at birth, as well as socio-economic and healthcare factors that influence disparities across different countries and regions. The dataset includes indicators such as GDP, immunization rates, and health expenditures. By using statistics, to find significant patterns and correlations that impact life expectancy worldwide.

#DATA WRANGLING


##Package import

In [69]:
import numpy as np
import pandas as pd

##Import Data

In [80]:
# Load the dataset
file_path = '/content/Life Expectancy at Birth.csv'  # Ensure this file is in the correct directory
life_expectancy_df = pd.read_csv(file_path)

In [81]:
df = pd.read_csv("/content/Life Expectancy at Birth.csv")


In [84]:
df.head()
#function is a method function that is already applied


ISO3               Country Continent           Hemisphere  \
0  AFG           Afghanistan      Asia  Northern Hemisphere   
1  AGO                Angola    Africa  Southern Hemisphere   
2  ALB               Albania    Europe  Northern Hemisphere   
3  AND               Andorra    Europe  Northern Hemisphere   
4  ARE  United Arab Emirates      Asia  Northern Hemisphere   

  Human Development Groups UNDP Developing Regions  HDI Rank (2021)  \
0                      Low                      SA            180.0   
1                   Medium                     SSA            148.0   
2                     High                     ECA             67.0   
3                Very High                     NaN             40.0   
4                Very High                      AS             26.0   

   Life Expectancy at Birth (1990)  Life Expectancy at Birth (1991)  \
0                          45.9672                          46.6631   
1                          41.8933                          43.8127   
2                          73.1439                          73.3776   
3                          78.4063                          77.9805   
4                          71.9004                          72.2414   

   Life Expectancy at Birth (1992)  ...  Life Expectancy at Birth (2012)  \
0                          47.5955  ...                          61.9230   
1                          42.2088  ...                          58.6231   
2                          73.7148  ...                          78.0640   
3                          80.3241  ...                          82.9166   
4                          72.3062  ...                          78.7155   

   Life Expectancy at Birth (2013)  Life Expectancy at Birth (2014)  \
0                          62.4167                          62.5451   
1                          59.3074                          60.0397   
2                          78.1226                          78.4075   
3                          82.9343                          82.9483   
4                          78.8528                          79.0442   

   Life Expectancy at Birth (2015)  Life Expectancy at Birth (2016)  \
0                          62.6587                          63.1361   
1                          60.6546                          61.0923   
2                          78.6441                          78.8602   
3                          82.9534                          82.9671   
4                          79.2232                          79.3347   

   Life Expectancy at Birth (2017)  Life Expectancy at Birth (2018)  \
0                          63.0160                          63.0810   
1                          61.6798                          62.1438   
2                          79.0473                          79.1838   
3                          82.9803                          82.9923   
4                          79.5036                          79.6274   

   Life Expectancy at Birth (2019)  Life Expectancy at Birth (2020)  \
0                          63.5645                          62.5751   
1                          62.4484                          62.2612   
2                          79.2825                          76.9893   
3                          83.0039                          79.0234   
4                          79.7262                          78.9457   

   Life Expectancy at Birth (2021)  
0                          61.9824  
1                          61.6434  
2                          76.4626  
3                          80.3684  
4                          78.7104  

[5 rows x 39 columns]

In [85]:
print(df.head())  # Show the first five rows
print(df.columns)  # Show all column names
print(df.info())  # Show dataset structure


  ISO3               Country Continent           Hemisphere  \
0  AFG           Afghanistan      Asia  Northern Hemisphere   
1  AGO                Angola    Africa  Southern Hemisphere   
2  ALB               Albania    Europe  Northern Hemisphere   
3  AND               Andorra    Europe  Northern Hemisphere   
4  ARE  United Arab Emirates      Asia  Northern Hemisphere   

  Human Development Groups UNDP Developing Regions  HDI Rank (2021)  \
0                      Low                      SA            180.0   
1                   Medium                     SSA            148.0   
2                     High                     ECA             67.0   
3                Very High                     NaN             40.0   
4                Very High                      AS             26.0   

   Life Expectancy at Birth (1990)  Life Expectancy at Birth (1991)  \
0                          45.9672                          46.6631   
1                          41.8933                  

In [72]:
# Import tech CSV file in "url" as a pandas DataFrame
# object assigned to the variable "df"

#class example code is df = pd.read_csv(url)


In [86]:
# Drop missing values to simplify analysis
life_expectancy_df.dropna(inplace=True)
# Compute summary statistics (mean, max, and median) for each numerical column
summary_statistics = life_expectancy_df.describe().T  # Transpose for readability

# Select only numerical columns for median calculation
numerical_cols = life_expectancy_df.select_dtypes(include=np.number).columns
summary_statistics['median'] = life_expectancy_df[numerical_cols].median()
summary_statistics['max'] = life_expectancy_df[numerical_cols].max()

In [87]:
df.head()
#function is a method function that is already applied

ISO3               Country Continent           Hemisphere  \
0  AFG           Afghanistan      Asia  Northern Hemisphere   
1  AGO                Angola    Africa  Southern Hemisphere   
2  ALB               Albania    Europe  Northern Hemisphere   
3  AND               Andorra    Europe  Northern Hemisphere   
4  ARE  United Arab Emirates      Asia  Northern Hemisphere   

  Human Development Groups UNDP Developing Regions  HDI Rank (2021)  \
0                      Low                      SA            180.0   
1                   Medium                     SSA            148.0   
2                     High                     ECA             67.0   
3                Very High                     NaN             40.0   
4                Very High                      AS             26.0   

   Life Expectancy at Birth (1990)  Life Expectancy at Birth (1991)  \
0                          45.9672                          46.6631   
1                          41.8933                          43.8127   
2                          73.1439                          73.3776   
3                          78.4063                          77.9805   
4                          71.9004                          72.2414   

   Life Expectancy at Birth (1992)  ...  Life Expectancy at Birth (2012)  \
0                          47.5955  ...                          61.9230   
1                          42.2088  ...                          58.6231   
2                          73.7148  ...                          78.0640   
3                          80.3241  ...                          82.9166   
4                          72.3062  ...                          78.7155   

   Life Expectancy at Birth (2013)  Life Expectancy at Birth (2014)  \
0                          62.4167                          62.5451   
1                          59.3074                          60.0397   
2                          78.1226                          78.4075   
3                          82.9343                          82.9483   
4                          78.8528                          79.0442   

   Life Expectancy at Birth (2015)  Life Expectancy at Birth (2016)  \
0                          62.6587                          63.1361   
1                          60.6546                          61.0923   
2                          78.6441                          78.8602   
3                          82.9534                          82.9671   
4                          79.2232                          79.3347   

   Life Expectancy at Birth (2017)  Life Expectancy at Birth (2018)  \
0                          63.0160                          63.0810   
1                          61.6798                          62.1438   
2                          79.0473                          79.1838   
3                          82.9803                          82.9923   
4                          79.5036                          79.6274   

   Life Expectancy at Birth (2019)  Life Expectancy at Birth (2020)  \
0                          63.5645                          62.5751   
1                          62.4484                          62.2612   
2                          79.2825                          76.9893   
3                          83.0039                          79.0234   
4                          79.7262                          78.9457   

   Life Expectancy at Birth (2021)  
0                          61.9824  
1                          61.6434  
2                          76.4626  
3                          80.3684  
4                          78.7104  

[5 rows x 39 columns]

In [88]:
#Show the "Contienent" colum
df['Continent']

0         Asia
1       Africa
2       Europe
3       Europe
4         Asia
        ...   
190    Oceania
191       Asia
192     Africa
193     Africa
194     Africa
Name: Continent, Length: 195, dtype: object

In [89]:
df.Country

0               Afghanistan
1                    Angola
2                   Albania
3                   Andorra
4      United Arab Emirates
               ...         
190                   Samoa
191                   Yemen
192            South Africa
193                  Zambia
194                Zimbabwe
Name: Country, Length: 195, dtype: object

In [90]:
df.Continent.unique()

array(['Asia', 'Africa', 'Europe', 'America', 'Oceania'], dtype=object)

In [91]:
df.iloc[0]
#this is a locator
#gives the first row of the data

ISO3                                               AFG
Country                                    Afghanistan
Continent                                         Asia
Hemisphere                         Northern Hemisphere
Human Development Groups                           Low
UNDP Developing Regions                             SA
HDI Rank (2021)                                  180.0
Life Expectancy at Birth (1990)                45.9672
Life Expectancy at Birth (1991)                46.6631
Life Expectancy at Birth (1992)                47.5955
Life Expectancy at Birth (1993)                51.4664
Life Expectancy at Birth (1994)                51.4945
Life Expectancy at Birth (1995)                52.5442
Life Expectancy at Birth (1996)                53.2433
Life Expectancy at Birth (1997)                53.6342
Life Expectancy at Birth (1998)                52.9431
Life Expectancy at Birth (1999)                54.8464
Life Expectancy at Birth (2000)                55.2978
Life Expectancy at Birth (2001)                55.7981
Life Expectancy at Birth (2002)                56.4538
Life Expectancy at Birth (2003)                57.3445
Life Expectancy at Birth (2004)                57.9436
Life Expectancy at Birth (2005)                58.3608
Life Expectancy at Birth (2006)                58.6844
Life Expectancy at Birth (2007)                59.1113
Life Expectancy at Birth (2008)                59.8515
Life Expectancy at Birth (2009)                60.3636
Life Expectancy at Birth (2010)                60.8508
Life Expectancy at Birth (2011)                61.4191
Life Expectancy at Birth (2012)                 61.923
Life Expectancy at Birth (2013)                62.4167
Life Expectancy at Birth (2014)                62.5451
Life Expectancy at Birth (2015)                62.6587
Life Expectancy at Birth (2016)                63.1361
Life Expectancy at Birth (2017)                 63.016
Life Expectancy at Birth (2018)                 63.081
Life Expectancy at Birth (2019)                63.5645
Life Expectancy at Birth (2020)                62.5751
Life Expectancy at Birth (2021)                61.9824
Name: 0, dtype: object

In [92]:
row, col = df.shape

In [93]:
df.iloc[0:5]
#row 0-5

ISO3               Country Continent           Hemisphere  \
0  AFG           Afghanistan      Asia  Northern Hemisphere   
1  AGO                Angola    Africa  Southern Hemisphere   
2  ALB               Albania    Europe  Northern Hemisphere   
3  AND               Andorra    Europe  Northern Hemisphere   
4  ARE  United Arab Emirates      Asia  Northern Hemisphere   

  Human Development Groups UNDP Developing Regions  HDI Rank (2021)  \
0                      Low                      SA            180.0   
1                   Medium                     SSA            148.0   
2                     High                     ECA             67.0   
3                Very High                     NaN             40.0   
4                Very High                      AS             26.0   

   Life Expectancy at Birth (1990)  Life Expectancy at Birth (1991)  \
0                          45.9672                          46.6631   
1                          41.8933                          43.8127   
2                          73.1439                          73.3776   
3                          78.4063                          77.9805   
4                          71.9004                          72.2414   

   Life Expectancy at Birth (1992)  ...  Life Expectancy at Birth (2012)  \
0                          47.5955  ...                          61.9230   
1                          42.2088  ...                          58.6231   
2                          73.7148  ...                          78.0640   
3                          80.3241  ...                          82.9166   
4                          72.3062  ...                          78.7155   

   Life Expectancy at Birth (2013)  Life Expectancy at Birth (2014)  \
0                          62.4167                          62.5451   
1                          59.3074                          60.0397   
2                          78.1226                          78.4075   
3                          82.9343                          82.9483   
4                          78.8528                          79.0442   

   Life Expectancy at Birth (2015)  Life Expectancy at Birth (2016)  \
0                          62.6587                          63.1361   
1                          60.6546                          61.0923   
2                          78.6441                          78.8602   
3                          82.9534                          82.9671   
4                          79.2232                          79.3347   

   Life Expectancy at Birth (2017)  Life Expectancy at Birth (2018)  \
0                          63.0160                          63.0810   
1                          61.6798                          62.1438   
2                          79.0473                          79.1838   
3                          82.9803                          82.9923   
4                          79.5036                          79.6274   

   Life Expectancy at Birth (2019)  Life Expectancy at Birth (2020)  \
0                          63.5645                          62.5751   
1                          62.4484                          62.2612   
2                          79.2825                          76.9893   
3                          83.0039                          79.0234   
4                          79.7262                          78.9457   

   Life Expectancy at Birth (2021)  
0                          61.9824  
1                          61.6434  
2                          76.4626  
3                          80.3684  
4                          78.7104  

[5 rows x 39 columns]

In [94]:
df.iloc[10]

ISO3                                               AZE
Country                                     Azerbaijan
Continent                                         Asia
Hemisphere                         Northern Hemisphere
Human Development Groups                          High
UNDP Developing Regions                            ECA
HDI Rank (2021)                                   91.0
Life Expectancy at Birth (1990)                62.3522
Life Expectancy at Birth (1991)                62.0524
Life Expectancy at Birth (1992)                61.2956
Life Expectancy at Birth (1993)                61.2856
Life Expectancy at Birth (1994)                61.5941
Life Expectancy at Birth (1995)                62.3052
Life Expectancy at Birth (1996)                62.6217
Life Expectancy at Birth (1997)                62.9728
Life Expectancy at Birth (1998)                63.5609
Life Expectancy at Birth (1999)                64.1892
Life Expectancy at Birth (2000)                64.8905
Life Expectancy at Birth (2001)                65.5101
Life Expectancy at Birth (2002)                66.2392
Life Expectancy at Birth (2003)                66.7563
Life Expectancy at Birth (2004)                67.3461
Life Expectancy at Birth (2005)                67.5619
Life Expectancy at Birth (2006)                68.1653
Life Expectancy at Birth (2007)                68.6032
Life Expectancy at Birth (2008)                68.8112
Life Expectancy at Birth (2009)                69.3556
Life Expectancy at Birth (2010)                69.5291
Life Expectancy at Birth (2011)                69.9835
Life Expectancy at Birth (2012)                70.3598
Life Expectancy at Birth (2013)                70.8814
Life Expectancy at Birth (2014)                71.1163
Life Expectancy at Birth (2015)                71.4512
Life Expectancy at Birth (2016)                72.0039
Life Expectancy at Birth (2017)                72.2478
Life Expectancy at Birth (2018)                72.7601
Life Expectancy at Birth (2019)                73.1023
Life Expectancy at Birth (2020)                66.8679
Life Expectancy at Birth (2021)                69.3658
Name: 10, dtype: object

In [95]:
df.iloc[0:5, 0:2]

ISO3               Country
0  AFG           Afghanistan
1  AGO                Angola
2  ALB               Albania
3  AND               Andorra
4  ARE  United Arab Emirates

In [96]:
df.loc[0:5, 'Country']

0             Afghanistan
1                  Angola
2                 Albania
3                 Andorra
4    United Arab Emirates
5               Argentina
Name: Country, dtype: object

In [97]:
df.loc[0:5, 'Country':'Continent']

Country Continent
0           Afghanistan      Asia
1                Angola    Africa
2               Albania    Europe
3               Andorra    Europe
4  United Arab Emirates      Asia
5             Argentina   America

In [20]:
df.loc[0:5, 'Continent']

0       Asia
1     Africa
2     Europe
3     Europe
4       Asia
5    America
Name: Continent, dtype: object

In [98]:
Continent = df.Continent

In [100]:
Continent

0         Asia
1       Africa
2       Europe
3       Europe
4         Asia
        ...   
190    Oceania
191       Asia
192     Africa
193     Africa
194     Africa
Name: Continent, Length: 195, dtype: object

In [101]:
#numpy array of continent
Continent.values

array(['Asia', 'Africa', 'Europe', 'Europe', 'Asia', 'America', 'Asia',
       'America', 'Oceania', 'Europe', 'Asia', 'Africa', 'Europe',
       'Africa', 'Africa', 'Asia', 'Europe', 'Asia', 'America', 'Europe',
       'Europe', 'America', 'America', 'America', 'America', 'Asia',
       'Asia', 'Africa', 'Africa', 'America', 'Europe', 'America', 'Asia',
       'Africa', 'Africa', 'Africa', 'Africa', 'America', 'Africa',
       'Africa', 'America', 'America', 'Asia', 'Europe', 'Europe',
       'Africa', 'America', 'Europe', 'America', 'Africa', 'America',
       'Africa', 'Africa', 'Europe', 'Europe', 'Africa', 'Europe',
       'Oceania', 'Europe', 'Oceania', 'Africa', 'Europe', 'Asia',
       'Africa', 'Africa', 'Africa', 'Africa', 'Africa', 'Europe',
       'America', 'America', 'America', 'Asia', 'America', 'Europe',
       'America', 'Europe', 'Asia', 'Asia', 'Europe', 'Asia', 'Asia',
       'Europe', 'Asia', 'Europe', 'America', 'Asia', 'Asia', 'Asia',
       'Africa', 'Asia', 'As

In [102]:
# tell me the unique elements in ""
df.Continent.unique()

array(['Asia', 'Africa', 'Europe', 'America', 'Oceania'], dtype=object)

In [103]:
Country = df.Country

In [104]:
Country

0               Afghanistan
1                    Angola
2                   Albania
3                   Andorra
4      United Arab Emirates
               ...         
190                   Samoa
191                   Yemen
192            South Africa
193                  Zambia
194                Zimbabwe
Name: Country, Length: 195, dtype: object

#Statistical Data for each Columns

In [105]:
# Display computed statistics
print("\nSummary Statistics:")
print(summary_statistics[['mean', 'max', 'median']])


Summary Statistics:
                                       mean       max     median
HDI Rank (2021)                  116.527027  191.0000  117.50000
Life Expectancy at Birth (1990)   61.678976   76.6301   63.75835
Life Expectancy at Birth (1991)   61.970728   76.5905   64.15725
Life Expectancy at Birth (1992)   61.890578   76.6269   64.21250
Life Expectancy at Birth (1993)   62.018374   76.7029   64.64135
Life Expectancy at Birth (1994)   62.322742   76.6700   64.96315
Life Expectancy at Birth (1995)   62.578247   76.5519   64.91715
Life Expectancy at Birth (1996)   62.759663   76.8042   65.29655
Life Expectancy at Birth (1997)   62.941059   77.2281   65.66315
Life Expectancy at Birth (1998)   62.923286   77.9875   65.49565
Life Expectancy at Birth (1999)   63.387561   78.4701   66.22450
Life Expectancy at Birth (2000)   63.912529   79.0691   66.42510
Life Expectancy at Birth (2001)   64.221391   79.1863   66.77215
Life Expectancy at Birth (2002)   64.551362   79.3025   67.02195
Life

In [107]:
 # Summary statistics for numerical each column
 print(life_expectancy_df.describe().T)

                                 count        mean        std      min  \
HDI Rank (2021)                  148.0  116.527027  44.305748  12.0000   
Life Expectancy at Birth (1990)  148.0   61.678976   9.504520  29.9353   
Life Expectancy at Birth (1991)  148.0   61.970728   9.445871  30.1479   
Life Expectancy at Birth (1992)  148.0   61.890578   9.729283  22.7245   
Life Expectancy at Birth (1993)  148.0   62.018374   9.868665  22.7041   
Life Expectancy at Birth (1994)  148.0   62.322742   9.893914  14.0985   
Life Expectancy at Birth (1995)  148.0   62.578247   9.489707  30.9483   
Life Expectancy at Birth (1996)  148.0   62.759663   9.550536  31.2966   
Life Expectancy at Birth (1997)  148.0   62.941059   9.592469  32.2424   
Life Expectancy at Birth (1998)  148.0   62.923286  10.092738  18.3852   
Life Expectancy at Birth (1999)  148.0   63.387561   9.585534  42.0712   
Life Expectancy at Birth (2000)  148.0   63.912529   9.255190  44.5181   
Life Expectancy at Birth (2001)  148.0

This gives the Life Expectancy at Birth through 1990-2021 with the count of each year being 148.0 with the  mean, std, and min.

In [116]:
 # Summary statistics
print(life_expectancy_df.describe())


       HDI Rank (2021)  Life Expectancy at Birth (1990)  \
count       148.000000                       148.000000   
mean        116.527027                        61.678976   
std          44.305748                         9.504520   
min          12.000000                        29.935300   
25%          80.000000                        55.589275   
50%         117.500000                        63.758350   
75%         154.250000                        69.512825   
max         191.000000                        76.630100   

       Life Expectancy at Birth (1991)  Life Expectancy at Birth (1992)  \
count                       148.000000                       148.000000   
mean                         61.970728                        61.890578   
std                           9.445871                         9.729283   
min                          30.147900                        22.724500   
25%                          55.594475                        55.043375   
50%               

In [114]:
# Display basic information about the dataset
print("Dataset Overview:")
print(life_expectancy_df.info())
# Provides column names, data types, and missing values


Dataset Overview:
<class 'pandas.core.frame.DataFrame'>
Index: 148 entries, 0 to 194
Data columns (total 39 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   ISO3                             148 non-null    object 
 1   Country                          148 non-null    object 
 2   Continent                        148 non-null    object 
 3   Hemisphere                       148 non-null    object 
 4   Human Development Groups         148 non-null    object 
 5   UNDP Developing Regions          148 non-null    object 
 6   HDI Rank (2021)                  148 non-null    float64
 7   Life Expectancy at Birth (1990)  148 non-null    float64
 8   Life Expectancy at Birth (1991)  148 non-null    float64
 9   Life Expectancy at Birth (1992)  148 non-null    float64
 10  Life Expectancy at Birth (1993)  148 non-null    float64
 11  Life Expectancy at Birth (1994)  148 non-null    float64
 12  Life Expe

Create a new dataframe for participants with depending on Continenet or Country and life expectancy at Birth at each year

In [109]:
#if continnent is africa and country is algeria
#show the life expectancy
df[(df['Continent'] == 'Africa') & (df['Country'] == 'Algeria')]

ISO3  Country Continent           Hemisphere Human Development Groups  \
49  DZA  Algeria    Africa  Northern Hemisphere                     High   

   UNDP Developing Regions  HDI Rank (2021)  Life Expectancy at Birth (1990)  \
49                      AS             91.0                           67.416   

    Life Expectancy at Birth (1991)  Life Expectancy at Birth (1992)  ...  \
49                          67.6875                          67.7571  ...   

    Life Expectancy at Birth (2012)  Life Expectancy at Birth (2013)  \
49                          74.2024                          74.6153   

    Life Expectancy at Birth (2014)  Life Expectancy at Birth (2015)  \
49                            75.11                           75.622   

    Life Expectancy at Birth (2016)  Life Expectancy at Birth (2017)  \
49                          75.7318                          75.7428   

    Life Expectancy at Birth (2018)  Life Expectancy at Birth (2019)  \
49                          76.0656                          76.4742   

    Life Expectancy at Birth (2020)  Life Expectancy at Birth (2021)  
49                          74.4528                          76.3767  

[1 rows x 39 columns]

According to my data, if a child was born in Africa in Algeria this the life expectancy at birth has increased since 1990 which was 67.4 to 2021 at 76.37

In [113]:
df[(df['Continent'] == 'Africa') & (df['Country'] == 'Algeria')] ['Life Expectancy at Birth (1990)']

49    67.416
Name: Life Expectancy at Birth (1990), dtype: float64

According to my data, if a child was born in African in Algeria there life Expectancy at Birth (1990) would be 67.416



In [117]:
df[(df['Continent'] == 'Asia') & (df['Country'] == 'China')] ['Life Expectancy at Birth (1999)']

32    71.4188
Name: Life Expectancy at Birth (1999), dtype: float64

According to my data, if a child was born in Asia, specifically China, in 1999, their life expectancy would be 71.4.

In [122]:
from IPython.display import display, HTML

display(HTML('<p style=\"color:blue; font-size:40px; font-weight:bold;\">Assignment 5 Week 8</p>'))

#**Quiz Questions**

1.   How do you import the pandas library in Python?

In [123]:
import pandas as pd

2.   How do you read a CSV file named data.csv into a pandas DataFrame?




In [ ]:
#Read a CSV file into a DataFrame:
df = pd.read_csv("data.csv")

3.  How do you display the first 5
 rows of a DataFrame named df?

In [ ]:
#Display the first 5 rows
df.head()

4. How do you find the number of rows and columns in a DataFrame df?



In [ ]:
#Find the number of rows and columns
df.shape

5. How do you print the names of all the columns in a DataFrame df?



In [ ]:
#Print column names
df.columns

6. How do you get the summary information of the DataFrame df including data types and non-null values?



In [ ]:
#Get summary information
df.info()

7. How do you select only the rows in DataFrame df where the column A is greater than 5
?

In [ ]:
#Select rows where cloumn A>5:
df[df['A']>5]

8. How do you select the columns A and B from the DataFrame df?



In [ ]:
#Select columns A and B:
df[['A', 'B']]

9. How do you rename the column A to Alpha in DataFrame df?



In [ ]:
#Rename column A to Alpha:
df.rename(columns={'A': 'Alpha'}, inplace=True)

10.  How do you replace all occurrences of the value 999
 in DataFrame df with nan?

In [ ]:
Replace 999 with NaN:
df.replace(999, np.nan, inplace=True)

11. How do you drop all rows in DataFrame df that contain missing values?



In [ ]:
#Drop rows with missing values:
df.dropna(inplace=True)

12. How do you fill all missing values in DataFrame df with 0
?

In [ ]:
#Fill missing values with 0:
df.fillna(0, inplace=True)

13.  How do you calculate the mean of column A in DataFrame df?



In [ ]:
#Calculate the mean of column A:
df['A'].mean()

14. How do you create a new column C in DataFrame df that is the sum of columns A and B?



In [ ]:
#Create a new column C as A + B:
df['C'] = df['A'] + df['B']

15. How do you save the DataFrame df to a CSV file named output.csv? (Find the solution by searching the pandas documentation.)



In [ ]:
#Save DataFrame to a CSV file:
df.to_csv("output.csv", index=False)